### ISSUE TITLE
Cannot add foreign key constraint (in import) even when using "Disable foreign keys" option

### ISSUE THREAD

**Issue by svivian** [phpmyadmin/phpmyadmin#13065]
### Steps to reproduce

1. Make two tables, one with a FK to the other (see below).
2. Export them from PMA, making sure to select the option "Disable foreign key checks"
3. Import back into the same database (with or without the option "Enable foreign key checks", it makes no difference)

### Expected behaviour

Data to be imported because foreign key checks should be turned off.

### Actual behaviour

I get an error:

	SQL query:

	CREATE TABLE `city` (
	  `id` int(11) UNSIGNED NOT NULL,
	  `name` varchar(100) NOT NULL
	) ENGINE=InnoDB DEFAULT CHARSET=latin1

	MySQL said:

	#1215 - Cannot add foreign key constraint

### Server configuration
**Operating system**: Mac or Ubuntu (occurs on both)

**Web server:** Apache or Nginx

**Database:** MySQL 5.6.13

**PHP version:** 5.5.38

**phpMyAdmin version:** 4.6.6 or 4.7 QA Github version (occurs on both)

### Client configuration
**Browser:** Chrome 56

**Operating system:** Mac OSX

### Data

Here's the export file of the test tables I created (minus comments):

    SET FOREIGN_KEY_CHECKS=0;
    SET SQL_MODE = "NO_AUTO_VALUE_ON_ZERO";
    SET time_zone = "+00:00";

    DROP TABLE IF EXISTS `city`;
    CREATE TABLE `city` (
      `id` int(11) UNSIGNED NOT NULL,
      `name` varchar(100) NOT NULL
    ) ENGINE=InnoDB DEFAULT CHARSET=latin1;

    INSERT INTO `city` (`id`, `name`) VALUES
    (1, 'Springfield'),
    (2, 'New New York');

    DROP TABLE IF EXISTS `property`;
    CREATE TABLE `property` (
      `id` int(10) UNSIGNED NOT NULL,
      `name` varchar(100) NOT NULL,
      `cityid` int(10) UNSIGNED NOT NULL
    ) ENGINE=InnoDB DEFAULT CHARSET=latin1;

    INSERT INTO `property` (`id`, `name`, `cityid`) VALUES
    (1, 'Simpsons House', 1),
    (2, 'Flanders House', 1),
    (3, 'Planet Express', 2);

    ALTER TABLE `city`
      ADD PRIMARY KEY (`id`);

    ALTER TABLE `property`
      ADD PRIMARY KEY (`id`),
      ADD KEY `cityid` (`cityid`);

    ALTER TABLE `city`
      MODIFY `id` int(11) UNSIGNED NOT NULL AUTO_INCREMENT, AUTO_INCREMENT=3;

    ALTER TABLE `property`
      MODIFY `id` int(10) UNSIGNED NOT NULL AUTO_INCREMENT, AUTO_INCREMENT=4;

    ALTER TABLE `property`
      ADD CONSTRAINT `property_ibfk_1` FOREIGN KEY (`cityid`) REFERENCES `city` (`id`);
    SET FOREIGN_KEY_CHECKS=1;

**Comment by svivian** [comment]
One possible way to solve this would be dropping all tables up front before recreating them. So instead of `DROP TABLE city; CREATE TABLE city...` we have `DROP TABLE city; DROP TABLE property; CREATE TABLE city...`

But it would be better to figure out exactly why `SET FOREIGN_KEY_CHECKS=0;` doesn't work.

**Comment by Achilles-96** [comment]
This is an error by MySQL itself.
@nijel ,
@ibennetch ,
Is it okay to modify the export file created by MySQL before exporting as @svivian suggested.

**Comment by svivian** [comment]
> Currently I don't recall what is actually reasons for creating indexes later and not on creating table...

You can't create a foreign key to a table that doesn't exist. If table A has a FK to table B and you're importing into a new database, adding table A then adding the FK would cause an error because B does not exist yet.

Also now that I think about it, dropping all tables up front will still fail if I am only importing the city table and not the property table.

**Comment by svivian** [comment]
Never mind, I realise you were talking about primary keys, not foreign keys. If I add `PRIMARY KEY (id)` to the `CREATE TABLE city` then it works as expected. I also can't think of any reasons to add primary/unique keys later on.

**Comment by nijel** [comment]
Speeding up data inserting can be done also by disabling keys for the insertion, this is what mysqldump does:

```
ALTER TABLE `property` DISABLE KEYS;
INSERT INTO `property` VALUES (1,'Simpsons House',1),(2,'Flanders House',1),(3,'Planet Express',2);
ALTER TABLE `property` ENABLE KEYS;
```

**Comment by madhuracj** [comment]
Indeed, that is correct. I can't think of any other reason. However, it's best to check the git history for any clues.

**Comment by nijel** [comment]
The `DISABLE KEYS` is quite limited, according to MySQL docs, "this option is effective only for nonunique indexes of MyISAM tables.".

So this is why there is good reason for creating indexes later as it will make imports run much faster than the ones from mysqldump (see discussion in https://github.com/phpmyadmin/phpmyadmin/issues/13319).
